pkg/cmd/server/origin/legacy.go

```diff
@@ -206,7 +207,8 @@ func LegacyStorage(storage map[schema.GroupVersion]map[string]rest.Storage) map[
 				case "deploymentConfigs":
 					restStorage := s.(*deploymentconfigetcd.REST)
 					store := *restStorage.Store
-					restStorage.DeleteStrategy = orphanByDefault(store.DeleteStrategy)
```

Comment:
@smarterclayton @mfojtik This seems to be indeed a bug that we have carried for a long time.

What do we want to do with legacy API cascade deletion? We enabled it for new DCs by https://github.com/openshift/origin/blob/e98a1a9bcf47784533bf76f2d1cae7659b5b4acd/pkg/deploy/controller/deploymentconfig/controller_ref_manager.go#L143 but fixing this bug seems to mitigate it. You can see it broken in our test using the legacy client: https://github.com/openshift/origin/blob/e98a1a9bcf47784533bf76f2d1cae7659b5b4acd/test/extended/deployments/deployments.go#L1098 https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install_update/4085/testReport/junit/(root)/Extended/deploymentconfigs__should_adhere_to_Three_Laws_of_Controllers__Conformance_/